diff --git a/libc/sysdeps/linux/arm/bits/arm_asm.h b/libc/sysdeps/linux/arm/bits/arm_asm.h
index 921c9a3..27d22bb 100644
--- a/libc/sysdeps/linux/arm/bits/arm_asm.h
+++ b/libc/sysdeps/linux/arm/bits/arm_asm.h
@@ -3,9 +3,11 @@
 #define _ARM_ASM_H
 
 #ifdef __thumb2__
+#ifdef __ASSEMBLER__
 .thumb
 .syntax unified
 #define IT(t, cond) i##t cond
+#endif /* __ASSEMBLER__ */
 #else
 /* XXX: This can be removed if/when we require an assembler that supports
    unified assembly syntax.  */
